database/src/main/dataset_schema/get_schema.sql

```diff
@@ -0,0 +1,202 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+DROP FUNCTION dataset_schema.get_schema(TEXT, INTEGER);
+DROP FUNCTION dataset_schema.get_schema(BIGINT);
+
+CREATE OR REPLACE FUNCTION dataset_schema.get_schema(
+    IN  i_schema_name       TEXT,
+    IN  i_schema_version    INTEGER DEFAULT NULL,
+    OUT status              INTEGER,
+    OUT status_text         TEXT,
+    OUT id_schema           BIGINT,
+    OUT schema_name         TEXT,
+    OUT schema_version      INTEGER,
+    OUT schema_description  TEXT,
+    OUT fields              JSONB,
+    OUT created_by          TEXT,
+    OUT created_when        TIMESTAMP WITH TIME ZONE,
+    OUT updated_by          TEXT,
+    OUT updated_when        TIMESTAMP WITH TIME ZONE,
+    OUT locked_by           TEXT,
+    OUT locked_when         TIMESTAMP WITH TIME ZONE,
+    OUT disabled_by         TEXT,
+    OUT disabled_when       TIMESTAMP WITH TIME ZONE
+) RETURNS record AS
+$$
+-------------------------------------------------------------------------------
+--
+-- Function: jobs_configuration.get_schema(2)
+--      Returns the data of the requested schema, based on its name and version
+--      If the version is omitted/NULL the latest version data are returned.
+--
+-- Parameters:
+--      i_schema_name       - name of the schema
+--      i_schema_version    - schema version to return, latest is taken if NULL
+--
+-- Returns:
+--      status              - Status code
+--      status_text         - Status text
+--      id_schema           - Id of the schema
+--      schema_name         - name of the schema
+--      schema_version      - the version of the schema
+--      schema_description  - description of the schema
+--      fields              - the fields the schema consist of
+--      created_by          - user who created the schema
+--      created_when        - time & date when the schema was disabled
+--      updated_by          - user who updated the schema to this particular version
+--      updated_when        - time & date when the this particular version of the schema was created
+--      locked_by           - if locked, who was the user who locked the schema
+--      locked_when         - if not NULL the schema is locked
+--      disabled_by         - if disabled, who was the user who disabled the schema
+--      disabled_when       - if not NULL the schema has been disabled
+--
+-- Status codes:
+--      200     - OK
+--      404     - Schema does not exist
+--      405     - Schema of the given version does not exist
```

Comment:
my vote here is to use std 404 code and just differentiate by the `status_text`
```suggestion
--      404     - Schema of the given version does not exist
```

Reply:
I understand that you want to distinguish the cases here, but I still believe that 404 is appropriate for both. 
The code describes entity requested is not found, how ever defined. So, 
`get_schema(existingSchema, notExistingVersion)` does not exist the same way as `get_schema(bogusSchema, regardlessVersion)`.

Btw. just schema existence, regardless of the version (something that that 405 should signify), can be checked by `get_schema(existingSchema, NULL)` if I read the code correctly.

`405` is Method Not Allowed (by which HTTP methods like GET, POST, ... is meant) so it does not seem to be fitting. So it would be nice to find an alternative status code with a proper meaning, but they don't just give out any `404B`'s :)